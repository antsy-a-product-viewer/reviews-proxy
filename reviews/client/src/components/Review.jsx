import React from 'react';
import Star from './Star.jsx';
import ReviewImg from './ReviewImg.jsx';
import moment from 'moment';
import _ from 'underscore';
import styles from './css/reviewStyles.css.js';

const Review = (props) => {
  

  return (
    <div>
      <div style={styles.info}>
        <div style={styles.username}>{props.review.user_name}</div>
        <div style={styles.date}>{moment(props.review.created_at).format('MMM DD, YYYY')}</div>
      </div>
      <div>
        <div>
          <Star starClass="review-star" stars={props.stars}/>
          <div>{props.review.review}</div>
          <ReviewImg style={{height: 300, width: 300, marginTop: 10}} review={props.review} imageClick={props.imageClick}/>
        </div>
      </div>
    </div>
  );
};

export default Review;