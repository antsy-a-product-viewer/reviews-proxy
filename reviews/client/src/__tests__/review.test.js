import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import Review from '../components/Review.jsx';

configure({ adapter: new Adapter() });

it('should render correctly with no props', () => {
  const props = {
    review: review
  };
  const component = shallow(<Review key={props.review.review_id} review={props.review}/>);
  
  expect(component).toMatchSnapshot();
});

it('should render as many stars as the review contained', () => {
  const props = {
    review: review
  };
  const component = mount(<Review key={props.review.review_id} review={props.review} stars={review.stars}/>);

  expect(component.find('.review-star')).toHaveLength(4);
});

const review = {
  'review_id': 35,
  'user_img': 'https://s3-us-west-1.amazonaws.com/antsyuserimages/00024.jpg',
  'user_name': 'Deangelo Schuppe',
  'created_at': '2014-03-27T15:37:49.000Z',
  'stars': 4,
  'review': 'I\'ll generate the cross-platform COM firewall, that should driver the XSS pixel!',
  'item_id': 91,
  'item_img': 'https://s3-us-west-1.amazonaws.com/antsyitemsimages/00091.jpg',
  'item_name': 'Rustic Metal Chair'
};
