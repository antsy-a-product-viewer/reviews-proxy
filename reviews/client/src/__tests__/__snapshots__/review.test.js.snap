// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "cursor": "pointer",
          "margin": 5,
          "textDecoration": "underline",
        }
      }
    >
      Deangelo Schuppe
    </div>
    <div
      style={
        Object {
          "margin": 5,
        }
      }
    >
      Mar 27, 2014
    </div>
  </div>
  <div>
    <div>
      <Star
        starClass="review-star"
      />
      <div>
        I'll generate the cross-platform COM firewall, that should driver the XSS pixel!
      </div>
    </div>
  </div>
</div>
`;
